L1Trigger/L1TTrackMatch/interface/L1TrackJetEmulationProducer.h

```diff
@@ -0,0 +1,77 @@
+#pragma once
+#include <iostream>
+#include <fstream>
+#include <cmath>
+#include <cstdlib>
+#include <string>
+#include <cstdlib>
+#include "DataFormats/L1Trigger/interface/TkJetWord.h"
+
+//For precision studies
+const int PT_EXTRABITS = 0;
+const int ETA_EXTRABITS = 0;
+const int PHI_EXTRABITS = 0;
+const int Z0_EXTRABITS = 0;
+
+typedef ap_ufixed<16 + PT_EXTRABITS, 11, AP_TRN, AP_SAT> pt_intern;
+typedef ap_int<14 + ETA_EXTRABITS> glbeta_intern;
+typedef ap_int<14 + PHI_EXTRABITS> glbphi_intern;
+typedef ap_int<10 + Z0_EXTRABITS> z0_intern;  // 40cm / 0.1
+
+namespace convert {
+  const int INTPHI_PI = 720;
+  const int INTPHI_TWOPI = 2 * INTPHI_PI;
+  static const float INTPT_LSB_POW = pow(2.0, -5 - PT_EXTRABITS);
+  static const float INTPT_LSB = INTPT_LSB_POW;
+  static const float ETA_LSB_POW = pow(2.0, -1 * ETA_EXTRABITS);
+  static const float ETA_LSB = M_PI / pow(2.0, 12) * ETA_LSB_POW;
+  static const float PHI_LSB_POW = pow(2.0, -1 * PHI_EXTRABITS);
+  static const float PHI_LSB = M_PI / pow(2.0, 12) * PHI_LSB_POW;
+  static const float Z0_LSB_POW = pow(2.0, -1 * Z0_EXTRABITS);
+  static const float Z0_LSB = 0.05 * Z0_LSB_POW;
+  inline float floatPt(pt_intern pt) { return pt.to_float(); }
+  inline int intPt(pt_intern pt) { return (ap_ufixed<18 + PT_EXTRABITS, 13 + PT_EXTRABITS>(pt)).to_int(); }
+  inline float floatEta(glbeta_intern eta) { return eta.to_float() * ETA_LSB; }
+  inline float floatPhi(glbphi_intern phi) { return phi.to_float() * PHI_LSB; }
+  inline float floatZ0(z0_intern z0) { return z0.to_float() * Z0_LSB; }
+
+  inline pt_intern makePt(int pt) { return ap_ufixed<18 + PT_EXTRABITS, 13 + PT_EXTRABITS>(pt); }
+  inline pt_intern makePtFromFloat(float pt) { return pt_intern(INTPT_LSB_POW * round(pt / INTPT_LSB_POW)); }
+  inline z0_intern makeZ0(float z0) { return z0_intern(round(z0 / Z0_LSB)); }
+
+  inline ap_uint<pt_intern::width> ptToInt(pt_intern pt) {
+    // note: this can be synthethized, e.g. when pT is used as intex in a LUT
+    ap_uint<pt_intern::width> ret = 0;
+    ret(pt_intern::width - 1, 0) = pt(pt_intern::width - 1, 0);
+    return ret;
+  }
+
+  inline glbeta_intern makeGlbEta(float eta) { return round(eta / ETA_LSB); }
+  inline glbeta_intern makeGlbEtaRoundEven(float eta) {
+    glbeta_intern ghweta = round(eta / ETA_LSB);
+    return (ghweta % 2) ? glbeta_intern(ghweta + 1) : ghweta;
+  }
+
+  inline glbphi_intern makeGlbPhi(float phi) { return round(phi / PHI_LSB); }
+
+};  // namespace convert
+
+//Each individual box in the eta and phi dimension.
+//  Also used to store final cluster data for each zbin.
+struct EtaPhiBin {
```

Comment:
There is already a class with this name in
https://github.com/cms-sw/cmssw/blob/522098e61e63196f7ac283966668910d88ecf188/L1Trigger/L1TTrackMatch/interface/L1TrackJetProducer.h#L10

Defining a different class here violates the One Definition Rule and results in undefined behavior. In general such generic type names should be avoided in the global namespace.